src/arch/mpi/conv-common.h

```diff
@@ -15,9 +15,9 @@
 #define CMK_HANDLE_SIGUSR                                  1
 
 #if CMK_ERROR_CHECKING
-#define CMK_MSG_HEADER_EXT_    CmiUInt2 rank, hdl,xhdl,info, redID; CmiInt4 root; unsigned char cksum, magic, mpiMsgType; CmiUInt1 zcMsgType:3, cmaMsgType:2;
+#define CMK_MSG_HEADER_EXT_    CmiUInt2 rank, hdl,xhdl,info, redID; CmiInt4 root; unsigned char cksum, magic, mpiMsgType; CmiUInt1 zcMsgType:3, cmaMsgType:2, readonly:1;
```

Comment:
'readonly' here and elsewhere in this patch might be slightly confusing since we already have Readonlies and Readonly Messages as concepts in Charm. I think it may be better to stick with nokeep?

Reply:
Yeah I guess for now sticking with nokeep makes more sense for now. Although I think it also might be worth changing nokeep at some point since it isn't particularly descriptive of what it actually does. Especially with the thought of potentially changing broadcasts to "readonly" by default in the future. But we can push that conversation until a time when it is more relevant.